metrics/internal/lv/space.go

```diff
@@ -52,6 +52,15 @@ func (s *Space) Reset() *Space {
 }
 
 func (s *Space) nodeFor(name string) *node {
+	// First check if the node is already present without taking write lock.
+	s.mtx.RLock()
+	n, ok := s.nodes[name]
+	s.mtx.RUnlock()
+	if ok {
+		return n
+	}
+
+	// Slow path, we need to create the node.
```

Comment:
now this method is not atomic - two identical `name`s can go and write into `s.nodes` - is that ok?

Reply:
i was pointing out that the method, previously, was atomic for a given `name`

now, with the lock code-flow, a single `name` could theoretically write into `s.nodes` - maybe it's not possible due to the nature of the code executing this method but from the scope of this method is is possible

and i was asking if it's ok that it now is possible

Reply:
> that it now is possible

I still don't understand what is now possible that wasn't before.

Reply:
ok i can finally see it, there is yet another check for `n, ok = s.nodes` so the second iteration with the same `name` won't write into `s.nodes`, i got lost in the code flow